doc/project/module.md

```diff
@@ -170,33 +228,84 @@ There are currently two module split styles: `FewestModules` and `SmallestModule
 
 #### `FewestModules`
 
-Create as few modules as possible without including unnecessary code.
+Create as few modules as possible
+* while respecting dynamic import boundaries and
+* without including unnecessary code.
+
 This is the default.
 
-In the example above, this would generate:
+In the entry points example above, this would generate:
 
 * `a.js`: public module, containing `AppA` and the export of `start`.
 * `b.js`: public module, containing `AppB`, `mutable.Set`, the export of `start` and the call to `AppB.main`
 * `a-b.js`: internal module, Scala.js core and the implementation of `println`.
 
 This also works for more than two public modules, creating intermediate shared (internal) modules as necessary.
 
+The dynamic import example above already assumes this module split style so a module listing is omitted.
+
 #### `SmallestModules`
 
 Create modules that are as small as possible.
-The smallest unit of splitting is a Scala class.
+The smallest unit of splitting is a Scala class (see [Splitting Granularity](#Splitting Granularity) below for more).
 
 Using this mode typically results in an internal module per class with the exception of classes that have circular dependencies: these are put into the same module to avoid a circular module dependency graph.
 
-In the example above, this would generate:
+In the entry points example above, this would generate:
 
 * `a.js`: public module, containing the export of `start`.
 * `b.js`: public module, containing the export of `start` and the call to `AppB.main`
 * many internal small modules (~50 for this example), approximately one per class.
 
+In the dynamic import example, this would generate:
+* `main.js`: public module, containing the export of `onClick`.
+* many internal small modules (~150 for this example), approximately one per class.
+
+Observe that in this split style, dynamic import boundaries do not affect splitting because the modules are already as small as possible.
```

Comment:
I don't understand this sentence. :s

Reply:
I have tried to clarify. PTAL

Reply:
(maybe it doesn't need to be there and I can just drop it)

Reply:
Hum, OK, now I understand. But I'm not sure it's clear for a user. Perhaps add something like "Dynamic import boundaries are still important for dynamic loading."? On a first reading, it sounds like this paragraph says that dynamic imports are irrelevant with SmallestModules. So it's confusing.

Perhaps even consider removing this paragraph altogether? I'm not sure it brings any relevant value to a user.